File: gaia-sdk-javascript/src/graphql/request/type/Tenant.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Represents tenant information
 */
export class Tenant extends Array<(_:VariableRegistry) => string> {
public _typeName = "Tenant";
    /**
     * The tenant id
     */
    public tenantId = () => { 
        this.push(_ => "tenantId")
    };

    /**
     * The name of the tenant
     */
    public qualifier = () => { 
        this.push(_ => "qualifier")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/Tenant.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class Tenant(list):
    """
    Represents tenant information
    """

    """
    The tenant id
    """
    def tenant_id(self):
        self.append(lambda x: "tenantId")

    """
    The name of the tenant
    """
    def qualifier(self):
        self.append(lambda x: "qualifier")

    """
    The list of implicit identities
    """
    def implicit_identities(self):
        self.append(lambda x: "implicitIdentities")

    """
    The list of explicit identities
    """
    def explicit_identities(self):
        self.append(lambda x: "explicitIdentities")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
